ingester/ingester.go

```diff
@@ -99,6 +101,21 @@ type userState struct {
 	ingestedSamples *ewmaRate
 }
 
+type flushOp struct {
+	from      model.Time
+	userID    string
+	fp        model.Fingerprint
+	immediate bool
+}
+
+func (o *flushOp) Key() string {
+	return fmt.Sprintf("%s-%d-%v", o.userID, o.fp, o.immediate)
+}
+
+func (o *flushOp) Priority() int64 {
+	return int64(o.from)
```

Comment:
Reviewer's note: https://github.com/weaveworks/cortex/commit/21b1b4caa625875332c0c92712b031e9fe8a7dc3, pushed straight to master, changes this to `-int64(o.from)`